x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/AllowWriteStep.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateObserver;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.Objects;
+
+/**
+ * Make shrunken index writable, if requested, by removing {@link IndexMetadata.APIBlock#WRITE } block on the index.
+ */
+public class AllowWriteStep extends UpdateSettingsStep {
+    public static final String NAME = "allow-write-after-shrink";
+
+    public static final Settings CLEAR_BLOCKS_WRITE_SETTING = Settings.builder()
```

Comment:
Nit: I find that `CLEAR_WRITE_BLOCK_SETTINGS` reads a bit better than using the verb also the setting**s** points out that this is not just one setting but the whole settings objects, which I think gives us the freedom to diverge a bit from the exact setting name. Just food for thought.